x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessHeaders.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc;
+
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.xpack.core.security.authc.CrossClusterAccessSubjectInfo;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public final class CrossClusterAccessHeaders {
```

Comment:
This is not a new class but rather a weirdly tracked diff: it's essentially what `RemoteAccessHeaders` was before this refactor.